src/cloudant/result.py

```diff
@@ -129,40 +130,80 @@ def type_or_none(typerefs, value):
     """
     return isinstance(value, typerefs) or value is None
 
+class ResultByKey(object):
+    """
+    Provides a wrapper for a value used to retrieving records from a result
```

Comment:
I think `to retrieving` should be changed to either `to retrieve` or `for retrieving`.